src/services/item/plugins/geolocation/repository.ts

```diff
@@ -0,0 +1,79 @@
+import { iso1A2Code } from '@rapideditor/country-coder';
+import { Between, EntityManager, Repository } from 'typeorm';
+
+import { AppDataSource } from '../../../../plugins/datasource';
+import { Item } from '../../entities/Item';
+import { ItemGeolocation } from './ItemGeolocation';
+
+export class ItemGeolocationRepository {
+  private repository: Repository<ItemGeolocation>;
+
+  constructor(manager?: EntityManager) {
+    if (manager) {
+      this.repository = manager.getRepository(ItemGeolocation);
+    } else {
+      this.repository = AppDataSource.getRepository(ItemGeolocation);
+    }
+  }
+
+  async getForItem(item: Item): Promise<ItemGeolocation | null> {
```

Comment:
I'm not sure that `getForItem` is the best name to describe the function, but maybe it is only me. For me, something like `getLocationByItem` or just `getByItem` is clearer.